framework/web/Cookie.php

```diff
@@ -65,4 +69,39 @@ public function __toString()
     {
         return (string) $this->value;
     }
+
+    /**
+     * Unserializes a cookie received by the client.
+     * @param string $data JSON encoded cookie value
+     * @return self|null
+     * @throws InvalidParamException if the data is not valid JSON
+     * @since 2.0.12
+     */
+    public static function fromDataString($data)
+    {
+        $unserialized = JSON::decode($data, true);
+        return \Yii::createObject([
+            'class' => get_called_class(),
+            'name' => $unserialized[0],
+            'value' => $unserialized[1],
```

Comment:
The data is validated first, you shouldn't pass unvalidated data into any function.